Cleanup repo layout

decriptor opened the pull request.

decriptor committed:
Restructure and convert to SDK style and net5

decriptor committed:
Code cleanup

decriptor committed:
Add ubuntu CI build

decriptor committed:
Run CI on all platforms

decriptor committed:
Add nuget.config. Should fix Windows build

trixtur commented:
Tried to run the debugger on the executable. I couldn't get the ConfigurationManager to read the config file anymore. Where does it need to go?

decriptor committed:
Clean up some variable naming

decriptor committed:
Add DI, initial serilog'ing, split app vs irc config

Added .idea to gitignore.

Started building Waving Hands.

Started building Waving Hands.

Fixe settings.

decriptor committed:
Add DI, initial serilog'ing, split app vs irc config

trixtur committed:
Added .idea to gitignore.

trixtur committed:
Started building Waving Hands

trixtur reviewed (changes requested):
We need to be able to pull the configuration from the web service.

decriptor commented on src/CsBot/IrcBotService.cs:
```diff
@@ -0,0 +1,352 @@
+		}
+
+		async Task<string> DownloadIrcConfig (Uri uri)
+		{
```
@trixtur Here is the code to download a configuration file. `WebClient` is dated and is replaced by httpclient

trixtur committed:
Added a test for the GetAt function.

decriptor committed:
Remove unneeded appsettings.json file

decriptor committed:
Merge branch 'cleanup-repo-layout' of github.com:trixtur/CsBot into cleanup-repo-layout

decriptor committed:
Touch up a couple things

decriptor committed:
Merge pull request #7 from trixtur/WavingHands

Starting building Waving hands game

trixtur committed:
Added a way to input the password.

trixtur committed:
Added await to async EventLoop and fixed json serialize issue.

trixtur reviewed (approved):
Loaded it up and ran some commands. Looks good!

decriptor committed:
Minor touchups, use SecureString instead (it's a start)

decriptor committed:
Also fix for server_validate in json
